lib/ex_crypto.ex

```diff
@@ -211,18 +211,63 @@ defmodule ExCrypto do
   """
   @spec encrypt(binary, binary, binary, binary) :: {:ok, {binary, {binary, binary, binary}}} | {:error, binary}
   def encrypt(key, authentication_data, initialization_vector, clear_text) do
-    case :crypto.block_encrypt(:aes_gcm, key, initialization_vector, {authentication_data, clear_text}) do
-      {cipher_text, cipher_tag} -> {:ok, {authentication_data, {initialization_vector, cipher_text, cipher_tag}}}
+    _encrypt(key, initialization_vector, {authentication_data, clear_text}, :aes_gcm)
+  end
+
+
+  @doc """
+  Encrypt a `binary` with AES in CBC mode.
+
+  Returns a tuple containing the `initialization_vector`, and `cipher_text`.
+
+  At a high level encryption using AES in CBC mode looks like this:
+
+      key + clear_text -> init_vec + cipher_text
+
+  ## Examples
+
+      iex> clear_text = "my-clear-text"
+      iex> {:ok, aes_256_key} = ExCrypto.generate_aes_key(:aes_256, :bytes)
+      iex> {:ok, {iv, payload}} = ExCrypto.encrypt(aes_256_key, clear_text)
+      iex> {iv, cipher_text} = payload
+      iex> assert(is_bitstring(cipher_text))
+      true
+
+  """
+  @spec encrypt(binary, binary, binary) :: {:ok, {binary, {binary, binary}}} | {:error, binary}
+  def encrypt(key, clear_text) do
```

Comment:
Yeah that sounds like a good idea...  Colin mentions "[m]any people suggested that I should instead recommend the use of AES in a block mode which provides both encryption and authentication, such as GCM, CCM, CWC, or EAX, and I explained some of my reasons for eschewing those options", but he's reccomending CTR over CBC which I've never used or seen used... but I imagine much the same logic applies?  

We can close or iterate on either or both of these PR's, I meant to work on fixing up this one over the weekend but forgot, maybe this week I'll have time.